src/Traits/ArrayTrait.php

```diff
@@ -18,12 +18,58 @@ protected function getFlatArray(array $array, string $delimiter, string $rootPre
         $result = [];
 
         foreach ($array as $key => $value) {
-            $new_key = $rootPrefix . (empty($rootPrefix) ? '' : $delimiter) . $key;
+            $newKey = $rootPrefix . (empty($rootPrefix) ? '' : $delimiter) . $key;
 
             if (is_array($value)) {
-                $result = array_merge($result, $this->getFlatArray($value, $delimiter, $new_key));
+                $result = array_merge($result, $this->getFlatArray($value, $delimiter, $newKey));
             } else {
-                $result[$new_key] = $value;
+                $result[$newKey] = $value;
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param array<mixed> $array
+     * @param string $delimiter
+     * @param string $rootPrefix
+     * @param int $lineNumber
+     *
+     * @return array<string, int>
+     */
+    protected function getLineNumbers(
+        array $array,
+        string $delimiter,
+        string $rootPrefix = '',
+        int $lineNumber = 0,
+        bool $closingBracket = false
+    ): array {
+        $result = [
+            '__LINE_NUMBER__' => $lineNumber
```

Comment:
ah is this the maximum? do we really need this?

Reply:
More or less. Its the current maximum so we have it even with recursion. Didn't know about a better way to keep the current line number and not lose it due to recursion...

Reply:
ok got it :)